dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgDataModel.ts

```diff
@@ -0,0 +1,1986 @@
+import { LksgData } from "@clients/backend";
+import { getDataset, DropdownDatasetIdentifier } from "@/utils/PremadeDropdownDatasets";
+import { Category } from "@/utils/GenericFrameworkTypes";
+
+export const lksgDataModel = [
+  {
+    name: "general",
+    label: "General",
+    color: "orange",
+    subcategories: [
+      {
+        name: "masterData",
+        label: "Master Data",
+        fields: [
+          {
+            showIf: (): boolean => true,
+            name: "dataDate",
+            description: "The date until when the information collected is valid",
+            label: "Data Date",
+            component: "DateFormField",
+            required: true,
+            validation: "required",
+          },
+          {
+            showIf: (): boolean => true,
+            name: "headOfficeInGermany",
+            description:
+              "Is your head office, administrative headquarters, registered office or subsidiary located in Germany?",
+            label: "Head Office in Germany",
+            component: "YesNoFormField",
+            required: true,
+            validation: "required",
+          },
+          {
+            showIf: (): boolean => true,
+            name: "groupOfCompanies",
+            description: "Do you belong to a group of companies?",
+            label: "Group of Companies",
+            component: "YesNoFormField",
+            required: true,
+            validation: "required",
+          },
+          {
+            showIf: (dataModel: LksgData): boolean => dataModel?.general?.masterData?.groupOfCompanies === "Yes",
+            name: "groupOfCompaniesName",
+            description: "What is the group of companies called?",
+            label: "Group of Companies Name",
+            component: "InputTextFormField",
+            required: true,
+            validation: "required",
+          },
+          {
+            showIf: (): boolean => true,
+            validation: "required|length:1",
+            name: "industry",
+            description: "In which industry is your company primarily active?",
+            label: "Industry",
+            component: "NaceCodeFormField",
+            required: true,
+          },
+          {
```

Comment:
We need to fix the validations:
![image](https://github.com/d-fine/Dataland/assets/47553592/b0c0b152-46df-4c36-bcf9-f84382441405)